File: Python3/2424. Longest Uploaded Prefix.py
class LUPrefix:

    def __init__(self, n: int):
        self.uploaded = [False] * n
        self.prefix = 0

    def upload(self, video: int) -> None:
        self.uploaded[video - 1] = True
        while self.prefix < len(self.uploaded) and self.uploaded[self.prefix]:
            self.prefix += 1

    def longest(self) -> int:
        return self.prefix


# Your LUPrefix object will be instantiated and called as such:
# obj = LUPrefix(n)
# obj.upload(video)
# param_2 = obj.longest()

File: Java/2424. Longest Uploaded Prefix.java
class LUPrefix {
    boolean[] uploaded;
    int prefix;

    public LUPrefix(int n) {
        uploaded = new boolean[n];
        Arrays.fill(uploaded, false);
        prefix = 0;
    }
    
    public void upload(int video) {
        uploaded[video - 1] = true;
        while (prefix < uploaded.length && uploaded[prefix]) {
            prefix++;
        }
    }
    
    public int longest() {
        return prefix;
    }
}

/**
 * Your LUPrefix object will be instantiated and called as such:
 * LUPrefix obj = new LUPrefix(n);
 * obj.upload(video);
 * int param_2 = obj.longest();
 */